src/main/java/com/libertymutualgroup/herman/aws/ecs/broker/s3/S3Broker.java

```diff
@@ -69,24 +90,39 @@ public S3Broker(S3CreateContext context) {
 
     public BucketMeta brokerFromConfigurationFile() {
         S3InjectConfiguration configuration = getS3Configuration();
-        AmazonS3 client = AmazonS3ClientBuilder.standard()
-            .withCredentials(new AWSStaticCredentialsProvider(context.getSessionCredentials()))
-            .withClientConfiguration(BambooCredentialsHandler.getConfiguration()).withRegion(context.getRegion()).build();
+        configuration.setEncryptionOption(configuration.getEncryptionOption() == null ?
+            taskProperties.getS3().getDefaultEncryption() : configuration.getEncryptionOption());
 
-        TagSet tags = new TagSet();
+        Map<String, String> tagMap = new HashMap<>();
         if (taskProperties != null) {
-            tags.setTag(taskProperties.getSbuTagKey(), configuration.getSbu());
-            tags.setTag(taskProperties.getOrgTagKey(), configuration.getOrg());
-            tags.setTag(taskProperties.getAppTagKey(), configuration.getAppName());
+            tagMap.put(taskProperties.getSbuTagKey(), configuration.getSbu());
+            tagMap.put(taskProperties.getOrgTagKey(), configuration.getOrg());
+            tagMap.put(taskProperties.getAppTagKey(), configuration.getAppName());
+        }
+
+        // Broker KMS key if required
+        if (S3EncryptionOption.KMS.equals(configuration.getEncryptionOption())
+                && Boolean.TRUE.equals(configuration.getCreateBucketKey())) {
+            configuration.setKmsKeyArn(brokerKms(configuration, tagMap));
         }
-        String policy = null;
 
+        // Setup up policy and tags for the bucket
+        String policy = null;
         if (configuration.getPolicyName() != null) {
             FileUtil fileUtil = new FileUtil(context.getRootPath(), buildLogger);
             policy = fileUtil.findFile(configuration.getPolicyName(), false);
         }
+        TagSet tags = new TagSet();
+        tagMap.entrySet().stream().forEach(it ->
```

Comment:
We do operations with tags pretty frequently, and TagSet isn't the greatest type to work with compared to HashMap. Definitely doesn't need to happen as part of this PR, but an idea for a future refactor: we may want to add some utility methods to do this Map -> TagSet conversion so we can work with Maps and then convert at the AWS touchpoint.